src/frame-writer.cpp

```diff
@@ -10,6 +10,14 @@
 #include <cstring>
 #include "averr.h"
 
+#include "config.h"
+
+#ifdef HAVE_OPENCL
+#include "opencl.hpp"
+
+std::unique_ptr<OpenCL> opencl;
```

Comment:
Can we not move this inside the `FrameWriter` class? Of course, this would also mean adding `#ifdef HAVE_OPENCL` checks to the `opencl.hpp` too

Reply:
You can probably even make the class empty when OpenCL is disabled (no members, just emtpy constructor) and then actually reduce the checks (for ex. you could remove the checks from meson in that case)

Reply:
I guess it's possible but for what reason? It just seems it would complicate the code.